js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -234,16 +235,51 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
     async sendTransaction(
         transaction: Transaction,
         connection: Connection,
-        _options?: SendOptions,
+        options?: SendOptions,
     ): Promise<TransactionSignature> {
         return await this.runWithGuard(async () => {
             const { authToken } = this.assertIsAuthorized();
             try {
                 return await this.transact(async (wallet) => {
-                    await this.performReauthorization(wallet, authToken);
+                    let targetCommitment: Finality;
+                    switch (connection.commitment) {
+                        case 'confirmed':
+                        case 'finalized':
+                        case 'processed':
+                            targetCommitment = connection.commitment;
+                            break;
+                        default:
+                            targetCommitment = 'finalized';
+                    }
+                    await Promise.all([
+                        this.performReauthorization(wallet, authToken),
+                        (async () => {
+                            if (transaction.recentBlockhash == null) {
+                                const preflightCommitment = options?.preflightCommitment;
+                                const preflightCommitmentScore =
+                                    preflightCommitment === 'finalized'
+                                        ? 2
+                                        : preflightCommitment === 'confirmed'
+                                        ? 1
+                                        : 0;
+                                const targetCommitmentScore =
+                                    targetCommitment === 'finalized' ? 2 : targetCommitment === 'confirmed' ? 1 : 0;
+                                const { blockhash } = await connection.getLatestBlockhash({
+                                    commitment:
+                                        preflightCommitment && preflightCommitmentScore < targetCommitmentScore
+                                            ? preflightCommitment
+                                            : targetCommitment,
+                                });
+                                transaction.recentBlockhash = blockhash;
+                            }
+                        })(),
+                    ]);
+                    if (transaction.feePayer == null && this.publicKey) {
+                        transaction.feePayer = this.publicKey;
+                    }
```

Comment:
```suggestion
                    transaction.feePayer = transaction.feePayer || this.publicKey;
```
It's going to fail anyway if neither feePayer or publicKey are set, so might as well get to it 😅

Style note -- do you prefer `||` or `??` if it doesn't matter (based on the types)? Is there a perf difference?

Reply:
Types is actually why I wrote it this way. `feePayer` doesn't accept `null` but `this.publicKey` might serve up `null`.

Here.

```typescript
transaction.feePayer ||= this.publicKey ?? undefined;
```